compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -205,6 +206,43 @@ pub(crate) fn create_object_file(sess: &Session) -> Option<write::Object<'static
         _ => elf::ELFOSABI_NONE,
     };
     let abi_version = 0;
+    // check bti protection
+    let mut check_cfprotection = false;
+    let mut pr_type: u32 = 0;
+    if architecture == Architecture::X86_64 && let rustc_session::config::CFProtection::Branch | rustc_session::config::CFProtection::Full = sess.opts.unstable_opts.cf_protection {
+      check_cfprotection =true;
+      pr_type = 0xc0000002;
+    }
+    if architecture == Architecture::Aarch64 && sess.opts.unstable_opts.branch_protection.is_some()
```

Comment:
Is there any downside to unconditionally adding the gnu properties for control flow/branch protection?

Reply:
I'm not sure how other architectures would react to this note being emitted. Here is another reference to BTI in the compiler  [context.rs](https://github.com/rust-lang/rust/blob/0a3b557d528dd7c8a88ceca6f7dc0699b89a3ef4/compiler/rustc_codegen_llvm/src/context.rs)

Reply:
That's an interesting thought. I don't think there's a downside actually, because there's no code to actually protect.

Reply:
on the LLVM side we can't do this just because we can't assume the support for the note section in the assemblers/linkers.
Here you can always emit the note section for the given architecture with the compatibility flag.

Reply:
OK, I changed the if statement to just check for the architecture.